File: Dynamic Programming/Medium/LET_no978_Longest Turbulent Subarray.cpp
class Solution {
public:
    int maxTurbulenceSize(vector<int>& arr) {
        int res = 0;
        for (int i = 0, count = 0; i + 1 < arr.size(); i++, count *= -1) {
            if (arr[i] > arr[i + 1])
                count = count > 0 ? count + 1 : 1;
            else if (arr[i] < arr[i + 1])
                count = count < 0 ? count - 1 : -1;
            else
                count = 0;
            res = max(res, abs(count));
        }
        return res + 1;
    }
};

File: Dynamic Programming/Medium/LET_no978_Longest Turbulent Subarray.py
class Solution:
    def maxTurbulenceSize(self, arr: List[int]) -> int:
        res = count = 0;
        for i in range(len(arr) - 1):
            if arr[i] > arr[i + 1]:
                count = count + 1 if count > 0 else 1
            elif arr[i] < arr[i + 1]:
                count = count - 1 if count < 0 else -1
            else:
                count = 0
            res = max(res, abs(count))
            count *= -1
        return res + 1